src/pages/Trainee/Components/EditDialog/EditDialog.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  DialogActions, Dialog, DialogContentText, DialogContent,
+  DialogTitle, Button, TextField, InputAdornment, makeStyles,
+} from '@material-ui/core';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import EmailIcon from '@material-ui/icons/Email';
+import * as yup from 'yup';
+
+export const useStyle = makeStyles(() => ({
+  margin: {
+    margin: '10px 0',
+  },
+}));
+
+const EditDialog = (props) => {
+  const {
+    open, onClose, onSubmit, defaultValues,
+  } = props;
+  const classes = useStyle();
+  const [state, setstate] = useState({
+    name: '', email: '',
+  });
+  const [onBlur, setBlur] = useState({});
+  const [schemaErrors, setSchemaErrors] = useState({});
+
+  const schema = yup.object().shape({
+    name: yup.string().required('Name is required').min(3, 'should have more then 3 characters'),
+    email: yup.string().required('Email is required').email(),
+  });
+
+  const handleErrors = (errors) => {
+    const schemaError = {};
+    if (Object.keys(errors).length) {
+      errors.inner.forEach((error) => {
+        schemaError[error.path] = error.message;
+      });
+    }
+    setSchemaErrors(schemaError);
+  };
+
+  const handleValidate = () => {
+    schema.validate(state, { abortEarly: false })
+      .then(() => { handleErrors({}); })
+      .catch((err) => { handleErrors(err); });
+  };
+
+  const handleBlur = (label) => {
+    setBlur({ ...onBlur, [label]: true });
+  };
+
+  const getError = (label) => {
+    if (onBlur[label]) {
+      return schemaErrors[label] || '';
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    handleValidate();
+  }, [state]);
+
+  const hasErrors = () => Object.keys(schemaErrors).length !== 0;
+
+  const isTouched = () => Object.keys(onBlur).length !== 0;
+
+  const handleInputField = (label, input) => {
+    setstate({
+      ...state, [label]: input.target.value,
+    });
+  };
+
+  const handleClose = () => {
+    setBlur({ Name: false, Email: false });
+    onClose();
+  };
+
+  const handleSubmit = (details) => {
+    setBlur({ Name: false, Email: false });
```

Comment:
why use state variable names in Pascal case always use camel case for declaring variables